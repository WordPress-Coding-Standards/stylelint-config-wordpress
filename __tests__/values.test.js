'use strict';

const fs = require( 'fs' ),
	config = require( '../' ),
	stylelint = require( 'stylelint' ),
	validCss = fs.readFileSync( './__tests__/values-valid.css', 'utf-8' ),
	invalidCss = fs.readFileSync( './__tests__/values-invalid.css', 'utf-8' );

describe( 'flags no warnings with valid values css', () => {
	let result;

	beforeEach( () => {
		result = stylelint.lint({
			code: validCss,
			config,
		});
	});

	it( 'did not error', () => {
		return result.then( data => (
			expect( data.errored ).toBeFalsy()
		) );
	});

	it( 'flags no warnings', () => {
		return result.then( data => (
			expect( data.results[0].warnings ).toHaveLength( 0 )
		) );
	});
});

describe( 'flags warnings with invalid values css', () => {
	let result;

	beforeEach( () => {
		result = stylelint.lint({
			code: invalidCss,
			config,
		});
	});

	it( 'did error', () => {
		return result.then( data => (
			expect( data.errored ).toBeTruthy()
		) );
	});

	it( 'flags correct number of warnings', () => {
		return result.then( data => (
			expect( data.results[0].warnings ).toHaveLength( 9 )
		) );
	});

	it( 'snapshot matches warnings', () => {
		return result.then( data => (
			expect( data.results[0].warnings ).toMatchSnapshot()
		) );
	});
});
