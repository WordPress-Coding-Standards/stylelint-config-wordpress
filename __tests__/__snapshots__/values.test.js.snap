// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flags warnings with invalid values css snapshot matches warnings 1`] = `
Array [
  Object {
    "column": 16,
    "line": 2,
    "rule": "declaration-block-trailing-semicolon",
    "severity": "error",
    "text": "Expected a trailing semicolon (declaration-block-trailing-semicolon)",
  },
  Object {
    "column": 13,
    "line": 2,
    "rule": "declaration-colon-space-after",
    "severity": "error",
    "text": "Expected single space after \\":\\" with a single-line declaration (declaration-colon-space-after)",
  },
  Object {
    "column": 15,
    "line": 12,
    "rule": "declaration-property-unit-allowed-list",
    "severity": "error",
    "text": "Unexpected unit \\"em\\" for property \\"line-height\\" (declaration-property-unit-allowed-list)",
  },
  Object {
    "column": 15,
    "line": 10,
    "rule": "font-family-name-quotes",
    "severity": "error",
    "text": "Expected quotes around \\"Times New Roman\\" (font-family-name-quotes)",
  },
  Object {
    "column": 15,
    "line": 11,
    "rule": "font-weight-notation",
    "severity": "error",
    "text": "Expected numeric font-weight notation (font-weight-notation)",
  },
  Object {
    "column": 11,
    "line": 6,
    "rule": "length-zero-no-unit",
    "severity": "error",
    "text": "Unexpected unit (length-zero-no-unit)",
  },
  Object {
    "column": 15,
    "line": 6,
    "rule": "length-zero-no-unit",
    "severity": "error",
    "text": "Unexpected unit (length-zero-no-unit)",
  },
  Object {
    "column": 24,
    "line": 6,
    "rule": "length-zero-no-unit",
    "severity": "error",
    "text": "Unexpected unit (length-zero-no-unit)",
  },
  Object {
    "column": 1,
    "line": 15,
    "rule": "no-duplicate-selectors",
    "severity": "error",
    "text": "Unexpected duplicate selector \\".selector\\", first used at line 1 (no-duplicate-selectors)",
  },
]
`;
